File: simpleadapter/src/main/java/com/github/deckyfx/simpleadapter/AdapterGroupItem.java
package com.github.deckyfx.simpleadapter;

/**
 * Created by decky on 5/21/15.
 */
public class AdapterGroupItem<T extends BaseItem> extends AdapterItem {
    public AdapterDataSet<T> childrens;

    public AdapterGroupItem() {
        this(new AdapterDataSet<T>());
    }

    public AdapterGroupItem(AdapterDataSet<T> dataset) {
        super();
        this.childrens = dataset;
    }
}

File: simpleadapter-kotlin/src/main/java/com/github/simpleadapter/AdapterGroupItem.kt
package com.itp.android.simpleadapter_kotlin

/**
 * Created by decky on 5/21/15.
 */
class AdapterGroupItem<T : AdapterItem> @JvmOverloads constructor(var childrens: AdapterDataSet<T> = AdapterDataSet<T>()) : AdapterItem()